### ISSUE TITLE
Having to set large offset

### ISSUE THREAD

**Issue by cnolimit** [jasonslyvia/react-lazyload#42]
I have been using **LazyLoad** for a while now and its been fine, i recently pulled my repo down and did a `npm install` and all of a sudden all my elements wrapped inside of the Lazy Load component don't work.

I have been hacking away and have come to find that only after setting a large offset can I then achieve the functionality that i once had.

Is there any particular reason why the further down the page i get the higher the offset I need to set. Is the parent element not the immediate parent to the **LazyLoad** component by default.

I have a set up as follows;

```
import Section from '../path/to/Section-Component'
...

<div className={'HomePage'}>
   <Section>
      <Section1 /> /* <= <LazyLoad> component inside of the Section1 component */
   </Section>
   <Section>
      <Section2 />
   </Section>
   <Section>
      <Section3 />
   </Section>
</div>

```

Within most Sections there will be a LazyLoad component, but the further down the page I get the higher the offset value.

is this normal behaviour?

**Comment by jasonslyvia** [comment]
Have you set `height` props for `<Lazyload>`?

**Comment by cnolimit** [comment]
yes, i did receive some console logs about this, but have since resolved them so every `<LazyLoad>` component has a height value, although it hasn't changed the behaviour of the offset.

**Comment by jasonslyvia** [comment]
Are these `height` props set properly? 

Let's say you have 10 `<Section>` in your page, if you set `height={10}` to each `<Lazyload>` contained in `<Section>`, those 10 section will be rendered (i.e. no lazyload effect) at once 

ps. it depends on your viewport height, but assuming it's larger than 100

**Comment by cnolimit** [comment]
Using that same example, if each `<Section>`'s  `<LazyLoad>` components height was set to 100 should you be able you visually see the change?

**Comment by jasonslyvia** [comment]
If your viewport height is 500, and using this setting will render first 5 <Section>, the remains will be rendered as `<div className="lazyload-placeholder">`

**Comment by cnolimit** [comment]
Okay that makes sense, i've made a design to better illustrate my scenario. 

<img width="844" alt="screen shot 2016-08-18 at 14 04 59" src="https://cloud.githubusercontent.com/assets/6928300/17774869/9b82f59c-654d-11e6-8f26-930fa742e33f.png">


What i want to know is in the image above is the LazyLoad component offset against the top of the viewport, or its parent container `<Section1 />`, `<Section2 />` ...
Because the lower the sections are to the top, the higher i'm having to set the offset it order for it to appear, otherwise it just doesn't display the image at all.

Side note: Once i've set all of the offsets, it seems like the LazyLoad becomes ineffective as the page is loading all of the images before scrolling to the location of the image.

**Comment by jasonslyvia** [comment]
By default it's against the viewport, but if you set `<Lazyload overflow>`, then it's against its overflow parent.

If you got unexpected behaviour, please check Lazyload's ancestors' style to find out if there's `overflow: auto;` set.

**Comment by cnolimit** [comment]
Hadn't realised i didn't respond to this.

I manage to partly resolve this issue by playing around with the overflow css values across the sections. Only I ran into another issue when setting the overflow prop to true, I have raised a separate issue for this here #48

**Comment by jasonslyvia** [comment]
let's move to #48
